functions/New-DbaAvailabilityGroup.ps1

```diff
@@ -350,11 +355,16 @@ function New-DbaAvailabilityGroup {
             $secondaries = @()
             foreach ($computer in $Secondary) {
                 try {
-                    $secondaries += Connect-SqlInstance -SqlInstance $computer -SqlCredential $SecondarySqlCredential
+                    $second = Connect-SqlInstance -SqlInstance $computer -SqlCredential $SecondarySqlCredential
                 } catch {
-                    Stop-Function -Message "Error occurred while establishing connection to $instance" -Category ConnectionError -ErrorRecord $_ -Target $Primary
+                    Stop-Function -Message "Error occurred while establishing connection to $computer" -Category ConnectionError -ErrorRecord $_ -Target $computer
+                    return
```

Comment:
Would we not want to continue here and just create the AG with the replicas that can be added?

Reply:
First of all: The diff is missleading. The "return" is not new, it was there before (look at the next "return", this was this "return") - so the command stopped, if one of the replicas was not available. And the same way I implemented the test for HADR - to stop the command.
Because as a user, I would like the command to stop here. I want to look at the reason for not beeing able to connect and I want to get the chance to activate HADR on the instance.
I think this should be the general way: If there are prerequeisits that are not met - stop the command. If only one element of elements piped to the command fails - carry on with the rest.

Reply:
I'm aware it was there before.

Reply:
Then I misinterpreted your comment - sorry!

Reply:
Would it be beneficial to check all the secondaries first, then say if one of 4 does not have it enabled, output that information; instead of stopping at the first one we find missing it? We could make this into a simple object with the details we've gathered so far, and show a Notes field (we do something like this in the copy commands) that has what is missing. Then spit out the error and that object if something is not found.

Seems like that would be more to the user's favor to know "hey out of these 4 two are missing HADR".

If we go that route it might be good to create a command like `Test-PreAgConfig` or something...could be a public function too I guess if we come up with a decent name.

Reply:
Good point. More info for the user, but more code for us. I would like to start with just setting a `$failed = true` and print a warning inside the loop and do the `Stop-Function` after the loop.

Reply:
Oh for sure. This iteration I think just check each one and then spit a warning out for each one found not enabled I think is a starting point. Test command can be discussed later, we should have an issue with a wish-list of commands that were going to be built somewhere.

Reply:
New problem: If I cannot connect to the primary, I fail the command immediately. What if I cannot connect to a secondary? Test all other secondaries first to report all connection problems? Or should we say: Failing to connect is always an immediate stop? What do other commands? I don't have your overview of the general design - so I need your help on this.

Reply:
Implemented instant failing on connection problems. Please have a look, @wsmelton